app/services/build_heritage.rb

```diff
@@ -57,22 +58,40 @@ def convert_params_for_model(new_params)
     new_params
   end
 
-  def sync_resources(attributes, resources, key_key)
-    resource_map = resources.pluck(key_key, :id).to_h
+  def sync_resources(attributes, curr_attributes, resource_key)
+    attr_map = curr_attributes.pluck(resource_key, :id).to_h
     attributes ||= []
+
     attributes.each do |attr|
-      attr[:id] = resource_map[attr[key_key]] if resource_map[attr[key_key]].present?
+      attr[:id] = attr_map[attr[resource_key]] if attr_map[attr[resource_key]].present?
     end
 
-    resources_to_delete = resource_map.keys - attributes.map { |attr| attr[key_key] }
+    resources_to_delete = attr_map.keys - attributes.map { |attr| attr[resource_key] }
     resources_to_delete.each do |k|
-      attributes << {id: resource_map[k], _destroy: '1'}
+      attributes << {id: attr_map[k], _destroy: '1'}
     end
     attributes
   end
 
+  def nullables
+    [:cpu]
+  end
+
+  def prenullify_params
+    nullifiers = []
+    heritage.services.each do |service|
+      nuller_hash = {}
+      nullables.each do |nullable|
+        nuller_hash[nullable] = nil
+      end
+      nullifiers << { id: service.id, **nuller_hash }
+    end
+    {"services_attributes" => nullifiers}
+  end
+
   def execute
     heritage.district = district if district.present?
+    heritage.attributes = prenullify_params
     heritage.attributes = params
     heritage
```

Comment:
`prenullify_params` seems to be replaced by `params`, what's the effect of `prenullify_params`?

Reply:
This actually performs a rails mass assignment. I expected this to be confusing. This `=` operator takes a hash and then assigns values into the attributes of the given object. See https://apidock.com/rails/ActiveModel/AttributeAssignment/attributes%3D

I will use the `assign_attributes` method instead to make this clearer. Thanks for pointing it out.